flytekit/core/promise.py

```diff
@@ -784,13 +787,24 @@ def __rshift__(self, other: Any):
     return Output(*promises)  # type: ignore
 
 
+T = typing.TypeVar("T")
+
+
 async def binding_data_from_python_std(
     ctx: _flyte_context.FlyteContext,
     expected_literal_type: _type_models.LiteralType,
     t_value: Any,
-    t_value_type: type,
+    t_value_type: typing.Type[T],
     nodes: List[Node],
 ) -> _literals_models.BindingData:
+    literal_type_override = None
+    transformer_override = None
+    if get_origin(t_value_type) is Annotated:
+        for annotation in get_args(t_value_type)[1:]:
+            if isinstance(annotation, TypeTransformer):
```

Comment:
should we even allow type transformers in annotations other than `FlytePickleTransformer`?